apps/judicial-system/api/src/app/modules/auth/auth.service.ts

```diff
@@ -32,18 +35,50 @@ export class AuthService {
   }
 
   async findDefender(nationalId: string): Promise<User | undefined> {
-    const res = await fetch(
-      `${this.config.backendUrl}/api/cases/limitedAccess/defender?nationalId=${nationalId}`,
-      {
-        headers: { authorization: `Bearer ${this.config.secretToken}` },
-      },
-    )
+    try {
+      const res = await fetch(
+        `${this.config.backendUrl}/api/cases/limitedAccess/defender?nationalId=${nationalId}`,
+        {
+          headers: { authorization: `Bearer ${this.config.secretToken}` },
+        },
+      )
+      if (res.ok) {
+        return await res.json()
+      }
+    } catch (error) {
+      console.warn('Defender not found', error)
+    }
 
-    if (!res.ok) {
-      return undefined
+    // If a defender doesn't have any active cases, we look them up
+    // in the lawyer registry because we want to at least display an empty
+    // case list for them to avoid confusion about them not having access to
+    // the judicial system
+    try {
+      const lawyerRegistryInfo = await this.defenderService.getLawyer(
+        nationalId,
+      )
+      if (lawyerRegistryInfo && lawyerRegistryInfo.nationalId === nationalId) {
+        return {
+          // Reason for this is so we trust the nationalId from the authentication provider
+          // just in case the lawyer registry does something strange, we don't want to create
+          // a user with a nationalId that is not the same as the one from the authentication provider
+          nationalId: nationalId,
+          name: lawyerRegistryInfo.name,
+          role: UserRole.DEFENDER,
+          email: lawyerRegistryInfo.email,
+          mobileNumber: lawyerRegistryInfo.phoneNr,
+          active: true,
+          title: 'verjandi',
+          id: uuid(),
+          created: new Date().toString(),
+          modified: new Date().toString(),
+        } as User
+      }
+    } catch (error) {
+      console.warn('Defender not found in lawyer registry', error)
```

Comment:
Consider logging as info for the same reason as above. Also, consider a more accurate message as the reason for the failure could be anything.